src/core/hard_forks.zig

```diff
@@ -0,0 +1,85 @@
+const std = @import("std");
+const ArrayList = std.ArrayList;
+const allocator = std.mem.Allocator;
+const Slot = @import("slot.zig").Slot;
+pub const HardFork = struct { slot: Slot, count: usize };
+pub const HardForks = struct {
+    hard_forks: ArrayList(HardFork),
+
+    const Self = @This();
+
+    pub fn default(alloc: allocator) Self {
+        return .{ .hard_forks = ArrayList(HardFork).init(alloc) };
+    }
+
+    pub fn register(self: *Self, new_slot: Slot) !void {
+        const index = for (self.hard_forks.items, 0..) |hard_fork, index| {
```

Comment:
can you change this to `maybe_index` and then instead of `!= null` check - use a `if (maybe_index) |index| {` - ill also add this to contributing.md